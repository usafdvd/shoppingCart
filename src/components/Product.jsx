import React from "react";
import { Card, CardContent, CardActions, Button, Typography, makeStyles, Hidden } from "@material-ui/core";
import { currencyFormatter } from "../helpers/currency";

const useStyles = makeStyles({
  root: {
    minWidth: 200,
    maxHeight: 400,
    minHeight: 350
  },
  title: {
    alignContent: "center"
  },
  body: {
    alignSelf: "end",
    textAlign: "center",
    maxHeight: 200,
    overflow: "auto"
  },
  actions: {
    display: "flex",
    justifyContent: "space-between"
  }
})

export default function Product({ item, addProductToCart }) {
  const classes = useStyles();
  // having to use dangerouslySet due to formatting coming from backend. Do not use for user input, open to XSS hacking
  const renderHTML = (rawHTML) => React.createElement("div", { dangerouslySetInnerHTML: { __html: rawHTML } });

  const handleClick = event => {
    event.preventDefault();
    addProductToCart(item);
  }

  return (
    <Card className={classes.root}>
      <CardContent>
        <Typography variant="h6" className={classes.title}>
          {item.name}
        </Typography>
        <Typography className={classes.body}>
          {renderHTML(item.description)}
        </Typography>
      </CardContent>
      <CardActions className={classes.actions}>
        <Hidden mdDown>
        <Typography>
          {item.vendor.name}
        </Typography>
        </Hidden>
        <Typography>
          {currencyFormatter(item.price, item.currency)}
        </Typography>
        <Button
          onClick={handleClick}
          data-testid={`add-product-${item.id}`}
          disabled={!item.in_stock}>
          Add to Cart
        </Button>
      </CardActions>
    </Card>
  )
}