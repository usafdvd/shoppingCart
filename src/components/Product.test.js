import TestRenderer from "react-test-renderer";
import Product from "./Product";

const mockProduct = {
  id: 1,
  name: "Test Product",
  description: "This is a short description",
  price: 22.0,
  currency: "USD",
  in_stock: true,
  vendor: {
    id: 18,
    name: "Colorado Rockies Co"
  }
}
const mockAddProduct = jest.fn();

it('render correctly text component', () => {  
  const ProductComponent = TestRenderer.create(<Product item={mockProduct} addProductToCart={mockAddProduct}/>).toJSON();
  expect(ProductComponent).toMatchSnapshot();
});

