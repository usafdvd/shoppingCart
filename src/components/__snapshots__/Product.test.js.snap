// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly text component 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiCardContent-root"
  >
    <h6
      className="MuiTypography-root makeStyles-title-2 MuiTypography-h6"
    >
      Test Product
    </h6>
    <p
      className="MuiTypography-root makeStyles-body-3 MuiTypography-body1"
    >
      <div
        dangerouslySetInnerHTML={
          Object {
            "__html": "This is a short description",
          }
        }
      />
    </p>
  </div>
  <div
    className="MuiCardActions-root makeStyles-actions-4 MuiCardActions-spacing"
  >
    <p
      className="MuiTypography-root MuiTypography-body1"
    >
      $22.00
    </p>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text"
      data-testid="add-product-1"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Add to Cart
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
